strawberry_django/utils/typing.py

```diff
@@ -37,10 +39,13 @@
 _Type = TypeVar("_Type", bound="StrawberryType | type")
 
 TypeOrSequence: TypeAlias = Union[_T, Sequence[_T]]
+TypeOrMapping: TypeAlias = Union[_T, Mapping[str, _T]]
 TypeOrIterable: TypeAlias = Union[_T, Iterable[_T]]
 UserType: TypeAlias = Union["AbstractBaseUser", "AnonymousUser"]
 PrefetchCallable: TypeAlias = Callable[[GraphQLResolveInfo], Prefetch]
 PrefetchType: TypeAlias = Union[str, Prefetch, PrefetchCallable]
+AnnotateCallable: TypeAlias = Callable[[GraphQLResolveInfo], BaseExpression]
+AnnotateType: TypeAlias = Union[BaseExpression, AnnotateCallable]
```

Comment:
I have to double-check if `BaseExpression` is the best or `Expression` is better. `BaseExpression` also allows `Subquery`, and I believe I have examples in some projects were we had to use `annotate` with `Subquery` because other annotation types didn't cover our use cases. I'll check.